bugsnag/src/main/java/com/bugsnag/MetaData.java

```diff
@@ -1,12 +1,22 @@
 package com.bugsnag;
 
+import java.io.Serializable;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.logging.Logger;
 
 class MetaData extends HashMap<String, Object> {
     private static final long serialVersionUID = 2530038179702722770L;
+    private static Logger logger = Logger.getLogger(MetaData.class.getName());
 
     public void addToTab(String tabName, String key, Object value) {
+        //check if the metadata to add is serializable
+        if (!(value instanceof Serializable)) {
```

Comment:
As discussed IRL, this doesn't mean the same as being serializable for Jackson to JSON. Let's take an alternative approach and make the serialiser in configuration public and docs/tests for customizing it to serialise a custom object.